src/metpy/io/text.py

```diff
@@ -0,0 +1,134 @@
+
+    return lon, lat
+
+
+def _regroup_lines(iterable):
+    starting_num = re.compile('^[0-9]')
+    lines = list(iterable)[::-1]
+    while lines:
+        line = lines.pop()
+        if not line.strip():
+            continue
+        parts = line.split()
+        while lines and starting_num.match(lines[-1]):
+            parts.extend(lines.pop().split())
+        yield parts
+
+
+def parse_wpc_surface_bulletin(bulletin, year=None):
+    """Parse a coded surface bulletin from NWS WPC into a GeoPandas GeoDataFrame.
+
+    Parameters
+    ----------
+    bulletin : str or file-like object
+        If str, the name of the file to be opened. If `bulletin` is a file-like object,
+        this will be read from directly.
+
+    Returns
+    -------
+    dataframe : pandas.DataFrame
+        A `DataFrame` where each row represents a pressure center or front. The `DataFrame`
+        has four columns: 'valid', 'feature', 'strength', and 'geometry'.
+
+    """
+    # Create list with lines of text from file
+    with contextlib.closing(open_as_needed(bulletin)) as file:
+        text = file.read().decode('utf-8')
+
+    parsed_text = []
+    valid_time = datetime.utcnow()
+    for parts in _regroup_lines(text.splitlines()):
+        # A single file may have multiple sets of data that are valid at different times. Set
+        # the valid_time string that will correspond to all the following lines parsed, until
+        # the next valid_time is found.
+        if parts[0] in ('VALID', 'SURFACE PROG VALID'):
+            dtstr = parts[-1]
+            valid_time = valid_time.replace(year=year or valid_time.year, month=int(dtstr[:2]),
+                                            day=int(dtstr[2:4]), hour=int(dtstr[4:6]),
+                                            minute=0, second=0, microsecond=0)
+        else:
+            feature, *info = parts
+            if feature in {'HIGHS', 'LOWS'}:
+                # For each pressure center, add its data as a new row
+                # While ideally these occur in pairs, some bulletins have had multiple
+                # locations for a single center strength value. So instead walk one at a time
+                # and keep track of the most recent strength.
+                strength = np.nan
+                for item in info:
+                    if len(item) <= 4:
+                        strength = int(item)
+                    else:
+                        parsed_text.append((valid_time, feature.rstrip('S'), strength,
+                                            Point(_decode_coords(item))))
+            elif feature in {'WARM', 'COLD', 'STNRY', 'OCFNT', 'TROF'}:
+                # Some bulletins include 'WK', 'MDT', or 'STG' to indicate the front's
+                # strength. If present, separate it from the rest of the info, which gives the
+                # position of the front.
+                if info[0][0] in string.ascii_letters:
+                    strength, *boundary = info
```

Comment:
Is there any consequence to the `Series.dtype` changing between `float` and `object` depending on the presence of these frontal strength strings?

Reply:
Hmmm, let me see what the experience is like dealing with this.

Reply:
So, I lean towards leaving Pandas just to handle the type of the strength column. As it currently stands, you can deal with it like:
```python
df.strength[df.feature == 'HIGH'].astype('f')
```
The alternatives I can think of are:
1. Convert frontal strength to numbers, like (-1, 0, 1) which we then have to document and starts diverging arbitrarily from the source data
2. Put them in separate columns, which seems like additional complexity for little gain